testcase/unittest/blockchain/votes/test_vote.py

```diff
@@ -0,0 +1,118 @@
+import os
+
+import pytest
+
+from loopchain.blockchain.types import ExternalAddress
+from loopchain.blockchain.types import Hash32
+from loopchain.blockchain.votes import v0_1a, v0_5
+from loopchain.blockchain.votes.vote import Vote
+from loopchain.crypto.signature import SignVerifier
+
+
+@pytest.mark.parametrize("version_name", ["v0_3", "v0_4"])
+def test_vote_v0_1a_equals_to(version_name: str):
+    import importlib
+    from loopchain.blockchain import votes
+
+    vote_version = importlib.import_module(f"{votes.__name__}.{version_name}")
+
+    assert v0_1a.BlockVote == vote_version.BlockVote
+    assert v0_1a.LeaderVote == vote_version.LeaderVote
+
+
+class _TestVoteBase:
+    SIGNERS = pytest.SIGNERS
+    SIGNER = pytest.SIGNERS[0]
+    REPS = pytest.REPS
+
+    @pytest.fixture
+    def vote(self, override_vote_factory) -> Vote:
+        pass
+
+    def test_origin_args_has_valid_form(self, vote: Vote):
+        assert "signature" in dir(vote)
+
+        args = vote.origin_args()
+        assert isinstance(args, dict)
+        assert "signature" not in args
+
+    def test_hash_check(self, vote: Vote):
```

Comment:
This test is dependent on vote.origin_args() method, i think we need concrete vote(real data, and this concrete vote must be satisfied requirements)